File: ember/oauth-google-ember/app/models/cell.js
import DS from 'ember-data';

const { attr, belongsTo, Model } = DS;

export default Model.extend({
  workbook:  belongsTo('workbook', {async: true}),
  worksheet: belongsTo('worksheet', {async: true}),
  row:       attr('number'),
  column:    attr('number'),
  value:     attr('string')
});

File: app/models/Cell.scala
package models

import play.api.libs.json._
import play.api.libs.functional.syntax._
import com.google.gdata.data.spreadsheet.CellEntry

case class Cell(
  id        : String = "",
  workbook  : String = "",
  worksheet : String = "",
  row       : Int = 0,
  column    : Int = 0,
  value     : String = ""
)

/**
 * Companion object for the Cell.
 * Just contains an implicit JSON formatter.
 */

object Cell {

  implicit val cellFormat = Json.format[Cell]

}
